### PR TITLE
Minor tweaks to utils and new TypeRegistry process

### PR DESCRIPTION
**Pull request by ahgittin**
I've been working on a new way of reading types from yaml -- PR coming in the next couple weeks -- but found these small improvements to the existing codebase which seemed appropriate and would be good to get in sooner.

### CODE REVIEW

**Comment by ahgittin** [comment]
this is pretty simple - @neykov @grkvlt could one of you cast a quick eye over it?

File: core/src/main/java/org/apache/brooklyn/core/typereg/RegisteredTypes.java [hunk 1]
```diff
@@ -151,7 +181,7 @@ public static RegisteredType addSuperType(RegisteredType type, @Nullable Registe
         return type;
     }
     @Beta
-    public static RegisteredType addSuperTypes(RegisteredType type, Iterable<Object> superTypesAsClassOrRegisteredType) {
+    public static RegisteredType addSuperTypes(RegisteredType type, Iterable<? extends Object> superTypesAsClassOrRegisteredType) {
```
Comment (aledsage) [inline core/src/main/java/org/apache/brooklyn/core/typereg/RegisteredTypes.java]: Just use `Iterable<?>` instead of `Iterable<? extends Object>`.

**Comment by aledsage** [comment]
@ahgittin (cc @neykov) I haven't looked thoroughly enough at the `BrooklynTypeRegistry` and `RegisteredType` before, to give a proper review of this.

It would be useful for me (and other contributors I'm sure) to get an overview of the type registry, and of your new way of reading types (we don't need to wait until the PR to get an overview - in fact posting to the dev@brooklyn mailing list for such big changes would be good).

File: utils/common/src/main/java/org/apache/brooklyn/util/exceptions/Exceptions.java [hunk 2]
```diff
@@ -461,4 +461,14 @@ public static String getMessageWithAppropriatePrefix(Throwable t) {
         return appendSeparator(getPrefixText(t), t.getMessage());
     }
 
+    /** Returns true if the root cause is a "boring" CNF, ie a straightforward declaration that the clazz is not found;
+     * this permits callers to include the cause only when it is interesting, ie caused by a dependent class not loadable.
+     */
+    public static boolean isRootBoringClassNotFound(Exception e, String clazz) {
+        Throwable root = Throwables.getRootCause(e);
+        if (!(root instanceof ClassNotFoundException)) return false;
```
Comment (neykov) [inline utils/common/src/main/java/org/apache/brooklyn/util/exceptions/Exceptions.java]: Include `NoClassDefFoundError`?

**Comment by neykov** [comment]
LGTM. Merging - can address changes in next PRs.

File: utils/common/src/main/java/org/apache/brooklyn/util/exceptions/Exceptions.java (hunk 2 above)
Comment (ahgittin) [inline utils/common/src/main/java/org/apache/brooklyn/util/exceptions/Exceptions.java]: possibly, at least if it has no useful message.  i think maybe sometimes that one can have an interesting message?  wait until we see it occurring without clutter.

**Comment by ahgittin** [comment]
@aledsage yes, i'd like to talk about that.  nearly have something cool to show.

**Pull request merged**